### PR TITLE
update the alert email signup form to support MSA

### PR DESCRIPTION
**Pull request by pnavarrc**
- Update the `Newsletter` component to use `AutocompleteRegions`
- Changed the component to use `setState` instead of using class attributes to store state
- Moved the email validation logic to a more reusable place

### Testing

- Subscribe to alerts to one or more locations
- Check that you are actually subscribed on Firebase (dev)

### CODE REVIEW

File: src/components/Newsletter/Newsletter.js [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 
 class Newsletter extends React.Component {
-  constructor() {
+  constructor(props) {
     super();
     this.form = null;
     this.emailInput = null;
-    this.alertsSelectionArray = [];
-    this.defaultValues = [];
     this.autocompleteOptions = getLocationNames();
```
Comment (ghop02) [inline src/components/Newsletter/Newsletter.js:15]: is this used anymore?

File: src/components/Newsletter/Newsletter.js [hunk 2]
```diff
@@ -145,41 +117,12 @@ class Newsletter extends React.Component {
...
-                variant="outlined"
-                {...params}
-                placeholder="+ Add alert locations"
-              />
-            )}
+          <AutocompleteRegions
+            regions={regions.all()}
+            selectedRegions={selectedRegions}
+            onChangeRegions={onChangeRegions}
+            ariaLabelledBy=""
```
Comment (ghop02) [inline src/components/Newsletter/Newsletter.js:124]: should this be empty?

**Review by ghop02 — APPROVED** [review]
Looks good!

File: src/components/Newsletter/Newsletter.js (hunk 1 above)
Comment (pnavarrc) [reply src/components/Newsletter/Newsletter.js]: good catch!

File: src/components/Newsletter/Newsletter.js (hunk 2 above)
Comment (pnavarrc) [reply src/components/Newsletter/Newsletter.js]: not really 😅  - I added a default label so it should be good even if we don't have a label for the element

**Comment by pnavarrc** [comment]
updated!

**Pull request merged**